UnityProject/Assets/Scripts/Weapons/WeaponNetworkActions.cs

```diff
@@ -94,9 +94,12 @@ public void CmdUnloadWeapon(GameObject weapon)
 		Ray2D ray = new Ray2D(transform.position, direction);
 		RpcShootBullet(weapon, ray.GetPoint(30f), bulletName, damageZone);
 
-		//TODO add a check to see if bullet or energy weapon
-		ItemFactory.SpawnItem(casingPrefab, transform.position, transform.parent);
-		if (!isFlashing)
+        if (wepBehavior.SpawnsCaseing)
```

Comment:
Where did you add this "spawnsCaseing flag"? I cant find it anywhere in your code. Or is it already there?